### DISCUSSION TITLE
Use graphql-ws instead of unmaintained subscriptions-transport-ws

### DISCUSSION THREAD

**Discussion by Mikastark** [the-guild-org/apollo-angular#1694] [Ideas]
Hello :)
First thanks for your amazing work on this wonderfull library. Now that apollo server version 3 has been released, it now come by default without subscriptions (see https://www.apollographql.com/docs/apollo-server/migration/#subscriptions). They implicitly encourage to not use subscriptions-transport-ws anymore as it's unmaintened and will be reworked someday. An actively maintained alternative could be graphql-ws but it doesn't use the same protocol.

It would be fantastic if apollo-angular could use graphql-ws instead of subscriptions-transport-ws as it can be a more efficient and ready for the future apollo vision of graphql subscriptions

That doesn't mean to fully drop subscriptions-transport-ws support of course but make graphql-ws compatible along it

**Comment by KeithGillette** [comment]
Yes, I think think the Apollo Angular documentation should be updated to use the Apollo-recommended `graphql-ws` WebSockets implementation. We've replaced `subscriptions-transport-ws` with `graphql-ws` in our Apollo Angular project following the [`graphql-ws` Apollo Client Web Recipe](https://github.com/enisdenjo/graphql-ws#recipes) but are still figuring out how to cleanly unsubscribe from all GraphQL Subscriptions on user sign-out, as `graphql-ws` doesn't seem to have an equivalent to the `subscriptions-transport-ws` `SubscriptionClient.unsubscribeAll()`.
